integration/sawtooth_integration/tests/test_config_smoke.py

```diff
@@ -0,0 +1,121 @@
+
+class TestConfigSmoke(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls._temp_dir = None
+
+        parent_parser = create_parent_parser('main')
+        cls._parser = argparse.ArgumentParser(parents=[parent_parser])
+        subparsers = cls._parser.add_subparsers(title='subcommands',
+                                                 dest='command')
+
+        config.add_config_parser(subparsers, parent_parser)
+
+        cls._ExpectedSettingResults = 'x: 1\ny: 1\n'
+
+    def setUp(self):
+        self._temp_dir = tempfile.mkdtemp()
+        self._data_dir = os.path.join(self._temp_dir, 'data')
+        os.makedirs(self._data_dir)
+
+        # create a wif key for signing
+        self._wif_file = os.path.join(self._temp_dir, 'test.wif')
+        with open(self._wif_file, 'wb') as wif:
+            wif.write(TEST_WIF.encode())
+
+    def _run(self, args):
+        try:
+            LOGGER.debug("Running %s", " ".join(args))
+            proc = subprocess.run(
+                args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+            LOGGER.debug(proc.stdout.decode())
+        except subprocess.CalledProcessError as err:
+            LOGGER.debug(err)
+            LOGGER.debug(err.stderr.decode())
+            traceback.print_exc()
+            self.fail(self.__class__.__name__)
+
+    def _parse_list_command(self):
+        cmd_args = ['config', 'settings', 'list', '--url', 'http://rest_api:8080']
+
+        return self._parser.parse_args(cmd_args)
+
+
+    def test_submit_then_list_settings(self):
+        ''' Test ability to list settings after submission of a setting.
+            Test submits a simple config transaction to the validator,
+            then confirms that the settings can be retrieved by the
+            command 'sawtooth config settings list', and that the retrieved
+            setting equals the input setting.
+            '''
+
+        print("test_list_settings")
+
+        # Submit transaction, then list it using subprocess
+        cmds = [
+            ['sawtooth','config', 'proposal', 'create', '-k', self._wif_file,
+             '--url', 'http://rest_api:8080', 'x=1', 'y=1'],
+            ['sawtooth', 'config', 'settings', 'list', '--url',
+             'http://rest_api:8080']
+        ]
+
+        for cmd in cmds:
+            self._run(cmd)
+
+        # Retrieve string of setting contents for comparison to input settings
+        args = self._parse_list_command()
+
+        # backup the environment
+        backup = sys.stdout
+        sys.stdout = StringIO()   # Capture the output of next statement
+        config.do_config(args)
+        settings = sys.stdout.getvalue() # release the output and store
+        sys.stdout.close()
+        # Restore the environment
+        sys.stdout = backup
+
+        self.assertEqual(settings, self._ExpectedSettingResults, 'Setting results did not match.' )
+
+def create_parent_parser(prog_name):
```

Comment:
We should do away with this style of running these tests:  we can just call the `main`  function directly from `sawtooth_cli.main` (see [poet/cli](https://github.com/hyperledger/sawtooth-core/blob/master/consensus/poet/cli/tests/test_genesis/tests.py#L65-L67) for an example)